src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -2686,15 +2675,16 @@ void BlockchainLMDB::add_max_block_size(uint64_t sz)
     throw0(DB_ERROR(lmdb_error("Failed to set max_block_size: ", result).c_str()));
 }
 
-
 std::vector<uint64_t> BlockchainLMDB::get_block_weights(uint64_t start_height, size_t count) const
 {
-  return get_block_info_64bit_fields(start_height, count, offsetof(mdb_block_info, bi_weight));
+  return get_block_info_64bit_fields(start_height, count,
+      [](const void* bi) { return static_cast<const mdb_block_info *>(bi)->bi_weight; });
```

Comment:
Done (it required renaming mdb_block_info_3 to just mdb_block_info because you apparently can't forward declare something and then later declare it as an alias).